### PR TITLE
Add ACUs

### LINKED ISSUE #5267: scrubbers can suck so hard they reduce atmospherics to below safe level
only seen it on the mining shuttle

### LINKED ISSUE #7330: Self-sufficient scrubbers don't seem to actually do anything.
### Environment
- Game Version: Staging 21082412
- OS Version: Windows 4121

### Detailed description of bug

Self-sufficient scrubbers such as those seen on shuttles don't seem to actually function in regards to removing gases. They don't filter out plasma, miasma, etc.

### Steps to reproduce
1. Release plasma onto a shuttle with a scrubber. 
2. Watch as the plasma never dissipates.

### PR DESCRIPTION
**Pull request by NoooneyDude**
CL: [Feature] Adds air alarms. Air alarms allow atmospheric technicians to control any connected scrubbers and vents.
Does not replace the existing air alarm prefab.
acu (bugs seen here are fixed)
CL: [Improvement] Scrubber and vent behaviour is improved. Closes #5267. Probably closes #7330.
CL: [New] Scrubbers, vents can now be welded. Floor tiles must now be removed to remove these devices to encourage welding to disable them.

> Thanks to Discordian `mtk` for creating the UI ACU casing.

### CODE REVIEW

**Comment by corp-0** [comment]
the GUI is amazing

File: UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		public AirController Controller;
+
+		private IAcuControllable device;
+
+		private void Awake()
+		{
+			device = GetComponent<IAcuControllable>();
+			if (device == null)
+			{
+				Logger.LogError($"{this} has no component that implements {nameof(IAcuControllable)}!");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs:37]:
```suggestion
				Logger.LogError($"{gameObject.Name} has no component that implements {nameof(IAcuControllable)}!");
```

Dont think `this` returns the object name?

File: UnityProject/Assets/Scripts/Objects/Atmospherics/AirController.cs [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+				SetOperatingMode(initialOperatingMode);
+			}
+		}
+
+		private GasMix GetAverageGasMix()
+		{
+			int sampleSourceCount = ConnectedDevices.Count + (acuSamplesAir ? 1 : 0);
+
+			// Sample the gas at the ACU (if enabled).
+			GasMix avgMix = GasMix.NewGasMix(acuSamplesAir ? facingMetaNode.GasMix : GasMixes.BaseEmptyMix);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Atmospherics/AirController.cs]: Might be a GC issue to create a new gas mix every 3 seconds for all AirAlarms, might wanna see if we can create one at spawn then clear and reuse.

File: UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs (hunk 1 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs:37]: It returns the component and the gameobject. IIRC.

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs:37]: ah

File: UnityProject/Assets/Scripts/Objects/Atmospherics/AcuSample.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+	/// <summary>
+	/// A single sample of the atmosphere for pressure, temperature and gas moles.
+	/// <remarks>See also related classes:<list type="bullet">
+	/// <item><description><seealso cref="AcuSample"/> represents a single sample of the atmosphere</description></item>
+	/// <item><description><seealso cref="AirController"/> Monitors the local atmosphere and controls related devices</description></item>
+	/// <item><description><seealso cref="AcuDevice"/> allows a connection to form between the <c>ACU</c> and devices</description></item>
+	/// </list></remarks>
+	/// </summary>
+	public class AcuSampleAverage
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Atmospherics/AcuSample.cs:60]: why does this need to be class/thing?
Wouldn't a direct reference to the gas mix or a gas mix class Work?

**Review by Bod9001 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Objects/Atmospherics/AcuSample.cs (hunk 3 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Atmospherics/AcuSample.cs:60]: Yep, that's how I originally had it. You can compare the two commits. I decided to create a new class to reduce the complexity in `AirController`. Piggybacking off `GasMix` just felt a little crude, too. Comes with a whole lot of baggage that isn't used.

**Pull request merged**